### PR TITLE
Delete a tag

### PR DESCRIPTION
**Pull request by rionshocker**
Implemented delete a tag feature as well as its tests.

### CODE REVIEW

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -133,6 +133,15 @@ public void addTag(Tag p) {
         tags.add(p);
     }
 
+    /**
+     * Deletes a {@code tag} from the address book.
+     * Deletes the tag from contacts using it as well.
+     */
+    public void deleteTag(Tag tag) {
```
Comment (Khaleelur-Rahman) [inline src/main/java/seedu/address/model/AddressBook.java:140]: Maybe the naming could be better, `tag` to `tagToBeDeleted`?

**Review by Khaleelur-Rahman — COMMENTED** [review]
Overall LGTM! Just a few naming changes :)

File: src/main/java/seedu/address/model/AddressBook.java (hunk 1 above)
Comment (rionshocker) [reply src/main/java/seedu/address/model/AddressBook.java:140]: Yep I agree! Thanks for spotting.

**Review by Khaleelur-Rahman — COMMENTED** [review]
LGTM!

**Pull request merged**